reverse_engineering/helpers/indexHelper.js

```diff
@@ -3,40 +3,49 @@ const restApiHelper = require('./restApiHelper');
 const clusterHelper = require('../../shared/helpers/clusterHelper');
 const parserHelper = require('./parserHelper');
 const { GET_META_REGEXP, GET_PARTITION_HASH_REGEXP, DEFAULT_NAME } = require('../../shared/constants');
+const { INDEX_TYPE } = require('../../shared/enums/indexType');
 
 const handleIndex = index => {
 	const indexData = getHackoladeCompatibleIndex(index);
 	return pickBy(indexData, value => !isUndefined(value));
 };
 
+const getIndexExpression = ({ index }) => {
+	if (index.index_key) {
+		return index.index_key.map(getExpression).join(',');
+	}
+};
+
 const getHackoladeCompatibleIndex = index => {
+	const indexExpression = getIndexExpression({ index });
+
 	if (index.is_primary) {
 		return {
 			indxName: index.name,
-			indxType: 'Primary',
+			indxType: INDEX_TYPE.primary,
 			usingGSI: index.using === 'gsi',
 		};
 	} else if (checkArrayIndex(index)) {
 		return {
 			indxName: index.name,
-			indxType: 'Array',
+			indxType: INDEX_TYPE.array,
 			usingGSI: index.using === 'gsi',
-			arrayExpr: index.index_key.map(getExpression).join(','),
+			arrayExpr: indexExpression,
 			whereClause: getWhereCondition(index),
 		};
 	} else if (checkMetaIndex(index)) {
 		return {
 			indxName: index.name,
-			indxType: 'Metadata',
-			metadataExpr: index.index_key.map(getExpression).join(','),
+			indxType: INDEX_TYPE.metadata,
+			metadataExpr: indexExpression,
```

Comment:
As you don't have to use `indexExpression` value in a few places at the same time it looks redundant to move it to a separate variable and create additional complexity. I would propose to call directly the `getIndexExpression({ index })` for each index type and it provides you don't need to use the if statement inside `getIndexExpression` function

Reply:
ok, let's leave that as is